src/main/java/cubicchunks/lighting/LightPropagator.java

```diff
@@ -147,8 +161,13 @@ public void propagateLight(BlockPos centerPos, Iterable<BlockPos> coords, ILight
                         // can't go further, the next block already has the same or higher light value
                         continue;
                     }
-                    blocks.setLightFor(type, nextPos, newLight);
-                    setLightCallback.accept(nextPos);
+                    if (blocks.setLightFor(type, nextPos, newLight)) {
+                        setLightCallback.accept(nextPos);
+                    } else {
+                        // If cube is not loaded we will notify neighbors so cube will update light when it loads.
+                        blocks.markEdgeNeedLightUpdate(pos, type);
```

Comment:
Why not actually go over all the neighbors?

Reply:
In other two cases we don't know where from light diffuse to missing cube. But in that certain case we do know that.